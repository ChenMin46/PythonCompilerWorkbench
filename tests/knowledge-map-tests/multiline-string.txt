"""John's cat liked "The Matrix"
but his dog
preferred the sequels."""
