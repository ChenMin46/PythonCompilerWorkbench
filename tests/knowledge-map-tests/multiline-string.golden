"""John's cat liked "The Matrix"
but his dog
preferred the sequels."""

foo = '''multi
    line
  string'''

bar = '''
multi
line
string
with empty
first line
'''

baz = """     
multi
line
string
with spaces in first line
"""

# This doesn't work, but it's nuts!!!
#"""multi
#    line
#      string
#        method
#          call""".join('''
#another
#  multi
#    line
#      string''')
===
Module
  body: [
    Expr
      value: Str ML>> '"""John\'s cat liked "The Matrix"\nbut his dog\npreferred the sequels."""'
        s: 'John\'s cat liked "The Matrix"\nbut his dog\npreferred the sequels.'
    Assign ML>> "foo = '''multi\n    line\n  string'''"
      targets: [
        Name >> 'foo'
          id: 'foo'
          ctx: Store
      ]
      value: Str ML>> "'''multi\n    line\n  string'''"
        s: 'multi\n    line\n  string'
    Assign ML>> "bar = '''\nmulti\nline\nstring\nwith empty\nfirst line\n'''"
      targets: [
        Name >> 'bar'
          id: 'bar'
          ctx: Store
      ]
      value: Str ML>> "'''\nmulti\nline\nstring\nwith empty\nfirst line\n'''"
        s: '\nmulti\nline\nstring\nwith empty\nfirst line\n'
    Assign ML>> 'baz = """     \nmulti\nline\nstring\nwith spaces in first line\n"""'
      targets: [
        Name >> 'baz'
          id: 'baz'
          ctx: Store
      ]
      value: Str ML>> '"""     \nmulti\nline\nstring\nwith spaces in first line\n"""'
        s: '     \nmulti\nline\nstring\nwith spaces in first line\n'
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "id_0", 
              "type": "Str", 
              "value": "\"\"\"John's cat liked \"The Matrix\"\nbut his dog\npreferred the sequels.\"\"\""
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "ctx": "Store", 
              "id": "id_0", 
              "type": "Name", 
              "value": "foo"
            }, 
            " ", 
            {
              "id": "tid_1", 
              "type": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "type": "Str", 
              "value": "'''multi\n    line\n  string'''"
            }
          ], 
          "id": "id_0", 
          "type": "Assign"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "ctx": "Store", 
              "id": "id_0", 
              "type": "Name", 
              "value": "bar"
            }, 
            " ", 
            {
              "id": "tid_2", 
              "type": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "type": "Str", 
              "value": "'''\nmulti\nline\nstring\nwith empty\nfirst line\n'''"
            }
          ], 
          "id": "id_0", 
          "type": "Assign"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "ctx": "Store", 
              "id": "id_0", 
              "type": "Name", 
              "value": "baz"
            }, 
            " ", 
            {
              "id": "tid_3", 
              "type": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "type": "Str", 
              "value": "\"\"\"     \nmulti\nline\nstring\nwith spaces in first line\n\"\"\""
            }
          ], 
          "id": "id_0", 
          "type": "Assign"
        }
      ], 
      "id": "id_0", 
      "type": "Module"
    }, 
    "\n\n", 
    {
      "id": "tid_4", 
      "type": "comment", 
      "value": "# This doesn't work, but it's nuts!!!"
    }, 
    "\n", 
    {
      "id": "tid_5", 
      "type": "comment", 
      "value": "#\"\"\"multi"
    }, 
    "\n", 
    {
      "id": "tid_6", 
      "type": "comment", 
      "value": "#    line"
    }, 
    "\n", 
    {
      "id": "tid_7", 
      "type": "comment", 
      "value": "#      string"
    }, 
    "\n", 
    {
      "id": "tid_8", 
      "type": "comment", 
      "value": "#        method"
    }, 
    "\n", 
    {
      "id": "tid_9", 
      "type": "comment", 
      "value": "#          call\"\"\".join('''"
    }, 
    "\n", 
    {
      "id": "tid_10", 
      "type": "comment", 
      "value": "#another"
    }, 
    "\n", 
    {
      "id": "tid_11", 
      "type": "comment", 
      "value": "#  multi"
    }, 
    "\n", 
    {
      "id": "tid_12", 
      "type": "comment", 
      "value": "#    line"
    }, 
    "\n", 
    {
      "id": "tid_13", 
      "type": "comment", 
      "value": "#      string''')"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "type": "ROOT_NODE"
}
